kart/working_copy/sqlserver.py

```diff
@@ -140,6 +140,32 @@ def _drop_spatial_index(self, sess, dataset):
         # SQL server deletes the spatial index automatically when the table is deleted.
         pass
 
+    def _initialise_sequence(self, sess, dataset):
+        start = dataset.feature_path_encoder.find_start_of_unassigned_range(dataset)
+        if start:
+            seq_name = f"{dataset.table_name}_{dataset.primary_key}_seq"
+            sess.execute(
+                f"CREATE SEQUENCE {self.table_identifier(seq_name)} START WITH {start};",
+            )
```

Comment:
this gets called for every working-copy write... won't `CREATE SEQUENCE` fail on a WC update where the sequence already exists?

Reply:
It only gets called during `write_full` which both creates and populates the table - and I try to delete it again when deleting the table. So, it's actually working fine. But there is still room for improvement:
1. I may as well make this a CREATE or ALTER depending on in the sequence exists - it's more robust in case something goes wrong / the user messes with the working copy in some way.
2. Just realised that the sequence should be updated every time you checkout a different commit, even in the case where that commit doesn't cause the table to be dropped + rewritten. This isn't done yet. Will see what's involved and add it to this PR or to a follow up PR.

Reply:
Done and done